services/consensusalgo/leanhelixconsensus/block_provider.go

```diff
@@ -0,0 +1,122 @@
+package leanhelixconsensus
+
+import (
+	"context"
+	"github.com/orbs-network/lean-helix-go"
+	lhprimitives "github.com/orbs-network/lean-helix-go/primitives"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/crypto/logic"
+	"github.com/orbs-network/orbs-network-go/crypto/signature"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/consensus"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/orbs-network/orbs-spec/types/go/services/handlers"
+)
+
+func (s *service) HandleBlockConsensus(ctx context.Context, input *handlers.HandleBlockConsensusInput) (*handlers.HandleBlockConsensusOutput, error) {
```

Comment:
why do you call your receiver `service`?

Reply:
The service implements `ConsensusBlockHandler`, no separate struct for it.
Also it is not a receiver - this func makes sure the block has valid consensus.  IMO it should have been named `ValidateBlockConsensus` (pending reply from Tal on that).

Reply:
receiver is the golang term for this: `func (s *service)`

I just don't feel that the consensus algo is a *service* per se - I think it's a pluggable component in the consensus service (which is actually the consensus context). WDYT?

Reply:
I consulted with OdedW on this, we would rather leave it the way it is now - consensus context and each of the consensus algos are separate services. The "consensus algo" is just a directory name, there is no such entity in the actual code, the underlying consensus algo implementations are the actual services.

Reply:
well, I disagree with @OdedWx and @talkol on this aspect - I think that when we're going to microservices, we'll have a single process for the consensus context with a thread for each consensus algo, rather than 3 processes.